apps/namada-interface/src/App/Staking/StakingOverview/AllValidatorsTable/AllValidatorsTable.tsx

```diff
@@ -0,0 +1,178 @@
+import { Table, TableLink, TableConfigurations } from "@namada/components";
+import { Validator } from "slices/StakingAndGovernance";
+import {
+  AllValidatorsSearchBar
+} from "./AllValidatorsTable.components";
+import { formatPercentage, assertNever } from "@namada/utils";
+import { useAppSelector, RootState } from "store";
+import { ValidatorsCallbacks } from "../StakingOverview";
+
+// AllValidators table row renderer and configuration
+// it contains callbacks defined in AllValidatorsCallbacks
+const AllValidatorsRowRenderer = (
+  validator: Validator,
+  callbacks?: ValidatorsCallbacks
+): JSX.Element => {
+  return (
+    <>
+      <td>
+        <TableLink
+          onClick={() => {
+            const formattedValidatorName = validator.name
+              .replace(" ", "-")
+              .toLowerCase();
+
+            callbacks && callbacks.onClickValidator(formattedValidatorName);
+          }}
+        >
+          {validator.name}
+        </TableLink>
+      </td>
+      <td>{validator.votingPower?.toString() ?? ""}</td>
+      <td>{formatPercentage(validator.commission)}</td>
+    </>
+  );
+};
+
+const getAllValidatorsConfiguration = (
+  navigateToValidatorDetails: (validatorId: string) => void,
+  onColumnClick: (column: AllValidatorsColumn) => void,
+  sort: Sort,
+): TableConfigurations<Validator, ValidatorsCallbacks> => {
+  const getLabelWithTriangle = (column: AllValidatorsColumn): string => {
+    let triangle = "";
+    if (sort.column === column) {
+      if (sort.ascending) {
+        triangle = " \u25b5"; // white up-pointing small triangle
+      } else {
+        triangle = " \u25bf"; // white down-pointing small triangle
+      }
+    }
+
+    return `${column}${triangle}`;
+  }
+
+  return {
+    rowRenderer: AllValidatorsRowRenderer,
+    callbacks: {
+      onClickValidator: navigateToValidatorDetails,
+    },
+    columns: [
+      {
+        uuid: "1",
+        columnLabel: getLabelWithTriangle(AllValidatorsColumn.Validator),
+        width: "45%",
+        onClick: () => onColumnClick(AllValidatorsColumn.Validator),
+      },
+      {
+        uuid: "2",
+        columnLabel: getLabelWithTriangle(AllValidatorsColumn.VotingPower),
+        width: "25%",
+        onClick: () => onColumnClick(AllValidatorsColumn.VotingPower),
+      },
+      {
+        uuid: "3",
+        columnLabel: getLabelWithTriangle(AllValidatorsColumn.Commission),
+        width: "30%",
+        onClick: () => onColumnClick(AllValidatorsColumn.Commission),
+      },
+    ],
+  };
+};
+
+const sortValidators = (sort: Sort, validators: Validator[]): Validator[] => {
+  const direction = sort.ascending ? 1 : -1;
+
+  const ascendingSortFn: (a: Validator, b: Validator) => number =
+    sort.column === AllValidatorsColumn.Validator ?
+      (a, b) => a.name.localeCompare(b.name) :
+    sort.column === AllValidatorsColumn.VotingPower ?
+      ((a, b) =>
+        a.votingPower === undefined || b.votingPower === undefined ? 0 :
```

Comment:
I think we could use `!a.votingPower || !b.votingPower` if we want a bit less code